android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/DropdownAnswer.kt

```diff
@@ -0,0 +1,98 @@
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.times
+import androidx.compose.ui.util.lerp
+import co.nimblehq.ic.kmm.suv.android.ui.theme.Typography
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.VerticalPager
+import com.google.accompanist.pager.calculateCurrentOffsetForPage
+import com.google.accompanist.pager.rememberPagerState
+import kotlin.math.absoluteValue
+
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+fun DropdownAnswer(
+    choices: List<String>,
+    onIndexChange: (Int) -> Unit
+) {
+    val padding = 56.dp
+    val pagerState = rememberPagerState()
+    LaunchedEffect(pagerState.currentPage) {
+        onIndexChange(pagerState.currentPage)
+    }
+    Box {
+        VerticalPager(
+            count = choices.size,
+            state = pagerState,
+            contentPadding = PaddingValues(vertical = padding),
+            modifier = Modifier
+                .height(170.dp)
+                .fillMaxWidth()
+        ) { page ->
+            val choice = choices[page]
+            val isHighlight = pagerState.currentPage == page
+            Box(
+                contentAlignment = Alignment.Center,
+                modifier = Modifier
+                    .height(56.dp)
+                    .fillMaxWidth()
+            ) {
+                Text(
+                    text = choice,
+                    color = Color.White,
+                    style = if (isHighlight) Typography.h6 else Typography.h6.copy(fontWeight = FontWeight.Normal),
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                    textAlign = TextAlign.Center,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .wrapContentSize()
+                        .graphicsLayer {
+                            val pageOffset = calculateCurrentOffsetForPage(page).absoluteValue
+                            alpha = lerp(
+                                start = 0.5f,
+                                stop = 1f,
+                                fraction = 1f - pageOffset.coerceIn(0f, 1f)
+                            )
+                        }
+                )
+            }
+        }
+        Divider(
+            color = Color.White,
+            thickness = 1.dp,
+            modifier = Modifier.padding(top = padding)
+        )
+        Divider(
+            color = Color.White,
+            thickness = 1.dp,
+            modifier = Modifier.padding(top = 2 * padding)
+        )
```

Comment:
Showing the divider like this seems hacky in my opinion, the implementation could lead to issue on different screen resolutions 🤔 Could we detect the visible item and show the divider?

Reply:
@hoangnguyen92dn Do you mean putting the divider in each item and then showing it like we are doing with the font style?
```kt
style = if (isHighlight) Typography.h6 else Typography.h6.copy(fontWeight = FontWeight.Normal),
```

I tried putting the divider into each page, but the result is not good enough. There will be animation on the divider:

https://user-images.githubusercontent.com/19943832/208647445-2ea6b0aa-f6b5-4932-b497-878bc94465e8.mov


I prefer to keep the current approach as the `Divider` will expand based on the parent view. I think we can focus on the parent view if there are issues with screen resolution. What do you think?

Reply:
@suho I see, Android doesn't support this kind of view originally. 

> I prefer to keep the current approach as the Divider will expand based on the parent view. I think we can focus on the parent view if there are issues with screen resolution.

🙆‍♂️ Deal 🤝 this approach looks even better.